src/lib/StateParser.cpp

```diff
@@ -0,0 +1,117 @@
+#include "StateParser.h"
+
+
+//
+TreeNode* StateParser::popHead(){
+	TreeNode* tempHead = mHeads.front();
+	mHeads.pop_front();
+
+	return tempHead;
+}
+
+
+
+//
+bool StateParser::isEmpty() const{
+    return mHeads.empty();
+}
+
+
+
+// Reads entire input to create complete program
+void StateParser::createForest(std::deque<Token> oInput){
+    while(oInput.front().text != "END"){
+        if(oInput.front().type == Type::END){ // for "{" and "}"
+            throw ParseError(oInput.front().line, oInput.front().column, oInput.front().text);
+        }
+
+        mHeads.push_back(createTree(oInput));
+    }
+}
+
+
+
+// Reads single expression or statement
+TreeNode* StateParser::createTree(std::deque<Token>& input){
+    if(isExp(input.front())){
+        New_Parser parser;
+        return parser.parseForState(input);
+    }
+
+    return createStatement(input);
+}
+
+
+// When given "if" or "while" creates a node with a condition and a true and false tree
+TreeNode* StateParser::createStatement(std::deque<Token>& input){
+    std::string stateStr = input.front().text;
+    input.pop_front(); // Reads statement command
+    std::unique_ptr<TreeStatement> stateHead(new TreeStatement(stateStr));
```

Comment:
any reason why you aren't using unique_ptr for StateParser? don't have to change anything if there aren't any memory issues, just curious